### PR TITLE
Add the capability of createIndex to take index expression in N1QL (i…

### PR DESCRIPTION
**Pull request by jianminzhao**
…n addition to JSON). #CBL-1939

### CODE REVIEW

File: LiteCore/Storage/KeyStore.cc [hunk 1]
```diff
@@ -66,12 +66,13 @@ namespace litecore {
         return seq;
     }
 
-    bool KeyStore::createIndex(slice name,
-                               slice expressionJSON,
-                               IndexSpec::Type type,
-                               const IndexSpec::Options* options)
+    bool KeyStore::createIndex2(slice name,
```
Comment (snej) [inline LiteCore/Storage/KeyStore.cc:69]: I think I'd prefer this be named just `createIndex`, since C++ allows overloading.

**Review by snej — CHANGES_REQUESTED** [review]
We need some more tests, too -- 
* of regular value indexes
* of using the C API.

File: LiteCore/Storage/KeyStore.cc (hunk 1 above)
Comment (borrrden) [reply LiteCore/Storage/KeyStore.cc:69]: Same for me!  This method already has some default parameters, the query language could be another so it doesn't even have to be overloaded if you add it at the end.  Overloading works too though.

**Review by borrrden — CHANGES_REQUESTED** [review]

**Review by pasin — COMMENTED** [review]
I don't have additional comments.

File: LiteCore/Storage/KeyStore.cc (hunk 1 above)
Comment (jianminzhao) [reply LiteCore/Storage/KeyStore.cc]: Changed accordingly.

**Review by snej — APPROVED** [review]

**Review by borrrden — APPROVED** [review]

**Pull request merged**